jetty-core/jetty-util/src/main/java/org/eclipse/jetty/util/URIUtil.java

```diff
@@ -1836,8 +1836,7 @@ public static URI toURI(String resource)
         Objects.requireNonNull(resource);
 
         // Only try URI for string for known schemes, otherwise assume it is a Path
-        ResourceFactory resourceFactory = ResourceFactory.getBestByScheme(resource);
```

Comment:
Oh I never noticed this getting a resource factory by scheme before?   This is not good, as now are we selecting a factory by lifecycle or by scheme?